### PR TITLE
cluster: add suspect regions cache

### LINKED ISSUE #2422: Region split during scheduling process is not handled properly
Consider we balance a region. The operator is _add learner + promote learner + remove peer_. If the region splits at the first 2 steps, there is only 1 region (which has the original region id) can continue removing the extra peer. Others will have an extra peer, until the replica-checker or rule-checker checks the region.

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve? 
Fix #2422

### What is changed and how it works?
- `raftCluster` maintains a cache, namely `suspectRegions`. It contains all regions that potentially broken and need to fix
- when region splits, add all sub regions to the cache (we can include more cases later, for example, operator timeout)
- when coordinator patrols all regions, it first checks suspect regions

### Check List 

Tests 
 - Unit test

Related changes
- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Release note 
- No release note

### CODE REVIEW

File: server/cluster/cluster_worker_test.go [hunk 1]
```diff
@@ -19,6 +19,9 @@ import (
 	"github.com/pingcap/kvproto/pkg/pdpb"
 
 	// Register schedulers
```
Comment (rleungx) [inline server/cluster/cluster_worker_test.go:21]: The comment should be moved

Comment (disksing) [reply server/cluster/cluster_worker_test.go:21]: ok

**Comment by disksing** [comment]
PTAL @rleungx @nolouch @lhy1024

File: server/cluster/coordinator.go [hunk 2]
```diff
@@ -103,6 +103,27 @@ func (c *coordinator) patrolRegions() {
 			return
 		}
 
+		// Check suspect regions first.
+		for _, id := range c.cluster.GetSuspectRegions() {
```
Comment (lhy1024) [inline server/cluster/coordinator.go:107]: it seems to be similar to the block below, can we abstract it into a function to avoid redundancy

Comment (disksing) [reply server/cluster/coordinator.go:107]: em, I think they are not quite alike. In this loop, the region may not exist yet, and we need to remove the id from suspect list in some cases.

**Pull request merged**